api/include/opentelemetry/logs/logger.h

```diff
@@ -55,25 +98,43 @@ class Logger
   virtual void EmitLogRecord(nostd::unique_ptr<LogRecord> &&log_record) noexcept = 0;
 
   /**
-   * Emit a Log Record object with custom span context
+   * Emit a Log Record object with arguments
    *
-   * @param log_record
+   * @param log_record Log record
+   * @tparam args Arguments
    */
-  virtual void EmitLogRecord(nostd::unique_ptr<LogRecord> &&log_record,
-                             const trace::SpanContext &trace_context)
+  template <class... ArgumentType>
+  void EmitLogRecord(nostd::unique_ptr<LogRecord> &&log_record, ArgumentType &&... args)
   {
     if (!log_record)
     {
       return;
     }
 
-    log_record->SetSpanId(trace_context.span_id());
-    log_record->SetTraceId(trace_context.trace_id());
-    log_record->SetTraceFlags(trace_context.trace_flags());
+    IgnoreTraitResult(
+        detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::template Set(
+            log_record.get(), std::forward<ArgumentType>(args))...);
 
     EmitLogRecord(std::move(log_record));
   }
 
+  /**
+   * Emit a Log Record object with arguments
+   *
+   * @param log_record
```

Comment:
Nit - we can remove log_record, and perhaps mention the allowed parameters (in any order).

Reply:
Done.